src/sentry/flags/providers.py

```diff
@@ -48,12 +50,83 @@ class InvalidProvider(Exception):
     ...
 
 
+class LaunchDarklyItemSerializer(serializers.Serializer):
+    accesses = serializers.ListField(required=True)
+    date = serializers.IntegerField(required=True)
+    member = serializers.DictField(required=True)
+    name = serializers.CharField(max_length=100, required=True)
+    description = serializers.CharField(required=True)
+
+
+"""
+LaunchDarkly has a lot more flag actions than what's in our
+ACTION_MAP. The "updated" action is the catch-all for actions
+that don't fit in the other buckets.
+"""
+
+
+def handle_launchdarkly_actions(action: str) -> int:
+    if action == "createFlag" or action == "cloneFlag":
+        return ACTION_MAP["created"]
+    if action == "deleteFlag":
+        return ACTION_MAP["deleted"]
+    else:
+        return ACTION_MAP["updated"]
+
+
+def handle_launchdarkly_event(
+    request_data: dict[str, Any], organization_id: int
+) -> list[FlagAuditLogRow]:
+    serializer = LaunchDarklyItemSerializer(data=request_data)
+    if not serializer.is_valid():
+        raise DeserializationError(serializer.errors)
+
+    result = serializer.validated_data
+
+    return [
+        {
+            "action": handle_launchdarkly_actions(access["action"]),
+            "created_at": datetime.datetime.fromtimestamp(result["date"] / 1000.0, datetime.UTC),
+            "created_by": result["member"]["email"],
+            "created_by_type": CREATED_BY_TYPE_MAP["email"],
+            "flag": result["name"],
+            "organization_id": organization_id,
+            "tags": {"description": result["description"]},
+        }
+        for access in result["accesses"]
+        if access["action"]
+        in (
+            "createFlag",
+            "cloneFlag",
+            "deleteFlag",
+            "updateFallthrough",
+            "updateOffVariation",
+            "updateOn",
+            "updateRules",
+            "updateScheduledChanges",
+            "updateDeprecated",
+            "updateFlagDefaultVariations",
+            "updateFlagVariations",
+            "updateGlobalArchived",
+            "updateRulesWithMeasuredRollout",
+            "updateFallthroughWithMeasuredRollout",
+            "updatePrerequisites",
+            "stopMeasuredRolloutOnFlagFallthrough",
+            "stopMeasuredRolloutOnFlagRule",
```

Comment:
@cmanallen these are the actions i thought were important

Reply:
Can we put these in a constant?